File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/files/FilesRevokePublicURLRequest.java
package com.github.seratch.jslack.api.methods.request.files;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class FilesRevokePublicURLRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `files:write:user`
     */
    private String token;

    /**
     * File to revoke
     */
    private String file;

    FilesRevokePublicURLRequest(String token, String file) {
        this.token = token;
        this.file = file;
    }

    public static FilesRevokePublicURLRequestBuilder builder() {
        return new FilesRevokePublicURLRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getFile() {
        return this.file;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setFile(String file) {
        this.file = file;
    }

    public static class FilesRevokePublicURLRequestBuilder {
        private String token;
        private String file;

        FilesRevokePublicURLRequestBuilder() {
        }

        public FilesRevokePublicURLRequest.FilesRevokePublicURLRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public FilesRevokePublicURLRequest.FilesRevokePublicURLRequestBuilder file(String file) {
            this.file = file;
            return this;
        }

        public FilesRevokePublicURLRequest build() {
            return new FilesRevokePublicURLRequest(token, file);
        }

        public String toString() {
            return "FilesRevokePublicURLRequest.FilesRevokePublicURLRequestBuilder(token=" + this.token + ", file=" + this.file + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/files/FilesRevokePublicURLRequest.kt
package work.socialhub.kslack.api.methods.request.files

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class FilesRevokePublicURLRequest(

    /** Authentication token. Requires scope: `files:write:user` */
    override var token: String?,
    /** File to revoke */
    var file: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("file", file)
        }
    }
}